### ISSUE TITLE
Invalid UTF-8 encoding written to database

### ISSUE THREAD

**Issue by nilmerg** [Icinga/icinga2#9308]
## Describe the bug

Adding a comment with non-ASCII chars in it results in the UI showing garbled text. This is due to Icinga 2 writing invalid UTF-8 to the database.

## To Reproduce

1. Make sure the charset for the IDO's resource is set to `lain1` in the configuration
2. Add this comment:
  ```
 中文; Zhōngwén
  ööääüü
  ```
3. You should see this result in the UI
![Screenshot from 2022-03-31 09-41-57](https://user-images.githubusercontent.com/16668527/161003606-85d7241d-c0df-4ac4-b066-e8637866daa8.png)

## Expected behavior
![Screenshot from 2022-03-31 09-44-51](https://user-images.githubusercontent.com/16668527/161004118-d80dd243-9eed-478d-865d-4117dba0d888.png)

## Your Environment

Include as many relevant details about the environment you experienced the problem in

* Version used (`icinga2 --version`): v2.13.0-247-g8c170855c
* Operating System and version: icinga/icinga2:master
* Enabled features (`icinga2 feature list`): api checker debuglog graphite icingadb ido-mysql ido-pgsql notification
* Icinga Web 2 version and modules (System - About): 2.10

## Additional context

What Icinga 2 writes to the database when adding the comment: `3F3F3B205A683F6E6777E96E0D0AF6F6F6F6F6F6F6`
What PHP wrote directly to the database when creating the expected screenshot: `E4B8ADE696873B205A68C58D6E6777C3A96E0D0AC3B6C3B6C3A4C3A4C3BCC3BC`

To let PHP write directly to the database, check out https://github.com/Icinga/icingaweb2/tree/umlaut-tests and open `/icingaweb2/umlaut/umlauts#!/icingaweb2/monitoring/list/comments`. Then add the comment into the text box and hit submit. The right column should refresh and all comments get the new data, but showing non-garbled and valid UTF-8.

This shows that the browser, Icinga Web 2, PHP and MySQL are correctly handling the input. Of course, when adding a comment the API and JSON encoding/decoding is also a part. Though, Web 2 doesn't use non-standard JSON encoding and what's sent to Icinga 2's API is also correctly encoded with unicode codepoints.

Looking at the debug log shows this:
```
[2022-03-31 07:41:53 +0000] debug/IdoMysqlConnection: Query: UPDATE icinga_comments SET author_name = 'icingaadmin',  comment_data = '中文; Zhōngwén\r\nööääüü',  comment_source = '1',  comment_time = FROM_UNIXTIME(1648712513),  comment_type = '2',  endpoint_object_id = 250,  entry_time = FROM_UNIXTIME(1648712513),  entry_time_usec = '112373',  entry_type = '1',  expiration_time = NULL,  expires = '0',  instance_id = 1,  internal_comment_id = '11',  is_persistent = '0',  name = 'dummy-101!ab41c4ae-d877-4dfe-a5f3-9b95f30458dc',  object_id = 893,  session_token = 1648707316 WHERE entry_time = FROM_UNIXTIME(1648712513) AND name = 'dummy-101!ab41c4ae-d877-4dfe-a5f3-9b95f30458dc' AND object_id = 893
```

which also looks correct. But invalid UTF-8 is still written to the IDO somehow.

**Comment by oskar456** [comment]
Is this on Debian Bullseye? I just upgraded and noticed the similar issue. It seems that [libmariadb3 package](https://packages.debian.org/bullseye/libmariadb3) has been [compiled with utf8mb4 client charset as default](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=933063). This breaks _garbage-in garbage-out_ principle of Icinga2 IDO MySQL, which pushes UTF8 characters into Latin1 tables and hopes that the database will not attempt to do any conversions.

I think this should be fixed on Icinga2 side to enforce MySQL connection encoding to be Latin1 so no conversions are made by the database.

**Comment by nilmerg** [comment]
Yes, debian bullseye. :bulb:

**Comment by lippserd** [comment]
> Hi @nilmerg, @Al2Klimov we have the same problem in RHEL8 and this bugfix fixes it. Are you planning to backport it? If yes, up to which version?
> 
> Thanks a lot

This is already included in the released versions 2.12.7 and 2.13.3.
